repository/src/main/java/com/epam/esm/repository/impl/CertificateRepositoryImpl.java

```diff
@@ -0,0 +1,109 @@
+package com.epam.esm.repository.impl;
+
+import com.epam.esm.entity.Certificate;
+import com.epam.esm.entity.Tag;
+import com.epam.esm.repository.CertificateRepository;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Root;
+import java.util.List;
+import java.util.Optional;
+
+@Repository
+public class CertificateRepositoryImpl implements CertificateRepository {
+
+    @Autowired
+    private SessionFactory manager;
+
+    @Override
+    public List<Certificate> getAllEntity(int limit, int offset) {
+        Session session = manager.getCurrentSession();
+        return session.createQuery("select c from Certificate c", Certificate.class)
+                .setMaxResults(limit)
+                .setFirstResult(offset)
+                .getResultList();
+    }
+
+    @Override
+    public Optional<Certificate> getEntityById(long id) {
+        Session session = manager.getCurrentSession();
+        return session
+                .createQuery("select c from Certificate c where c.id = :id order by c.id ", Certificate.class)
+                .setParameter("id", id).uniqueResultOptional();
+    }
+
+    @Override
+    public void addEntity(Certificate certificate) {
+        Session session = manager.getCurrentSession();
+        session.save(certificate);
+    }
+
+    public void deleteEntity(Certificate certificate) {
+        Session session = manager.getCurrentSession();
+        session.delete(certificate);
```

Comment:
Do you delete entity from "certificates_tag" table in this operation?

Reply:
yes, I delete entity from "certificates_tag" table in this operation.